src/common/common-globalconf/src/main/java/ee/ria/xroad/common/conf/globalconf/FSGlobalConfValidator.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * The MIT License
+ * Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package ee.ria.xroad.common.conf.globalconf;
+
+import ee.ria.xroad.common.SystemProperties;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+@Slf4j
+public class FSGlobalConfValidator {
+
+    public GlobalConfInitState getReadinessState(String globalConfigurationDir) {
+        var globalConfPathOpt = resolveGlobalConfDir(globalConfigurationDir);
+        if (globalConfPathOpt.isEmpty()) {
+            log.warn("Invalid path: {}", globalConfigurationDir);
+            return GlobalConfInitState.FAILURE_CONFIGURATION_ERROR;
+        }
+
+        try {
+            var globalConfPath = globalConfPathOpt.get();
+            if (!isDir(globalConfPath)) {
+                log.warn("GlobalConf at [{}] is not a directory. Initialization will not continue.", globalConfPath);
+                return GlobalConfInitState.FAILURE_CONFIGURATION_ERROR;
+            }
+            if (!isAnchorPresent()) {
+                log.warn("GlobalConf Anchor file is missing. Initialization will not continue.");
```

Comment:
Consider using the term "Configuration Anchor" instead of "GlobalConf Anchor" since "Configuration Anchor" is the term that's used in the documentation.

> GlobalConf Anchor file is missing.